frontend/src/pages/_app.js

```diff
@@ -34,17 +34,27 @@ import './_app.scss'
 import AuthProvider from '@contexts/AuthContext'
 import { bool, func, object } from 'prop-types'
 
-function KitspaceApp({ Component, pageProps, session, isStaticFallback }) {
-  const setSession = session ? (
+function KitspaceApp({
+  Component,
+  pageProps,
+  serverSideSession,
+  clientSideSession,
+  isStaticFallback,
+}) {
+  const setSession = serverSideSession ? (
     <script
+      // using dangerouslySetInnerHTML to avoid browser parsing errors
+      // (could be that these only happen in development mode)
       // eslint-disable-next-line react/no-danger
       dangerouslySetInnerHTML={{
-        __html: `window.session = ${JSON.stringify(session)};`,
+        __html: `window.session = ${JSON.stringify(serverSideSession)};`,
```

Comment:
Isn't this redundant? Previously the values of `user` and `csrf` were read from `window.session` now it's passed directly to the context.

Reply:
I thought so too at first and tried to remove it, but it's still needed. It makes sense when you think about it. The only way to for the client-side to get the session. when you use `getInitialProps`, is for the server to embed it into the page. 

I can see a cleaner solution by using `getServerSideProps` on every page to get the session from gitea, rather than us a custom _app.js (we can make something re-usable like we did for `withRequireSignIn)`, but let's leave that for another PR.